### ISSUE TITLE
Private symbols by default and move public symbols to private

### ISSUE THREAD

**Issue by mgrauer** [girder/girder#2387]
This is a placeholder issue because "private symbols by default" is a process issue, and "move public symbols to private" along with an audit of public symbols are both open ended.

**Comment by danlamanna** [comment]
@brianhelba had mentioned having a programmatic way of checking that PRs don't accidentally introduce new symbols.

One way of approaching this would be diffing the list of public symbols, a list of public symbols could be crudely obtained via:
```
ctags -f - -R --languages=python --python-kinds=-i girder | awk '$1 !~ /^_/ && $NF !~ /^function:/'
```

**Comment by danlamanna** [comment]
@brianhelba introduced provisional work for Python in #2447, but JavaScript symbol tracking still needs to be addressed.

**Comment by mgrauer** [comment]
Next Steps: Split this into two or more issues, one to move public symbols to private in order to reduce the public API surface, and have that surface become well defined including specifying behavior (maybe be a third issue), another issue to add support to CI to flag new public symbols (with whitelist of public symbols that allows CI to pass)--or accomplish this same end in a different way.

**Comment by jeffbaumes** [comment]
@kotfic is this ok to close now that #2814 is merged?

**Comment by danlamanna** [comment]
There's still a component to this that needs to be done before Girder 3
which is to evaluate the current public API surface and determine what
needs to be made private.
